### DISCUSSION TITLE
doris的label的理解。

### DISCUSSION THREAD

**Discussion by coolback** [apache/doris#6675] [A - General / Q&A] [answered]
大佬们 doris里面的label是个什么概念，可以看到表里有哪些label吗

**Comment by morningman** [comment, marked as answer]
label 是用于表示一次导入操作的标识。在doris中，任何导入方式都会有一个label。这个label可以是用户指定的，也可能由系统自动生成。
label的目的是帮助实现“最多一次”的导入语义。即同一个label，在一个database内，只能被成功导入一次。如果一个label已经成功导入，如果再次使用这个label进行导入，会报错“Label already used”。而如果导入失败，则可以使用相同的label继续重试导入。

通过“最多一次”的导入语义，如果配合上游系统“至少一次”的数据生产语义，则可以实现端到端的“Exactly-once”语义，保证数据不丢不重。

无法查看一个表里有哪些label，因为label对应的是导入任务，而不是表。

label is an identifier used to indicate an load operation. In Doris, any load method will have a label. This label can be specified by the user or automatically generated by the system.
The purpose of label is to help achieve "at most once" semantics. That is, the same label can only be successfully loaded once in a database. If a label has been successfully loaded, if you use this label again to load, an error "Label already used" will be reported. If the load fails, you can continue to retry the load using the same label.

Through the load semantics of "at most once", if it is combined with the "at least once" data production semantics of the upstream system, end-to-end "Exactly-once" semantics can be realized to ensure that data is not lost or duplicated.

Cannot view which labels are in a table, because the label corresponds to the load task, not the table.

**Reply by zealinux** [reply]
你好, 文中所说 Doris Label【可由系统自动生成】，这样的 SQL改怎样写？


```sql
    LOAD LABEL example_db.exmpale_label_1
    (
        DATA INFILE("s3://your_bucket_name/your_file.txt")
        INTO TABLE load_test
        COLUMNS TERMINATED BY ","
    )
    WITH S3
    (
        "AWS_ENDPOINT" = "AWS_ENDPOINT",
        "AWS_ACCESS_KEY" = "AWS_ACCESS_KEY",
        "AWS_SECRET_KEY"="AWS_SECRET_KEY",
        "AWS_REGION" = "AWS_REGION"
    )
    PROPERTIES
    (
        "timeout" = "3600"
    );
```

**Comment by coolback** [comment]
感谢大佬指点，非常感谢

Sent from ProtonMail for iOS

在 周四, 9月 16, 2021 在 10:58 下午, Mingyu Chen ***@***.***> 来信：
